Source/DafnyCore/ProofDependencyWarnings.cs

```diff
@@ -81,8 +86,13 @@ public class ProofDependencyWarnings {
         return false;
       }
 
-      // Similarly here
-      if (poDep.ProofObligation is MatchIsComplete or AlternativeIsComplete) {
+      // Some proof obligations occur in a context that the Dafny programmer
+      // doesn't have control of, so warning about vacuity isn't helpful.
+      if (poDep.ProofObligation
```

Comment:
I think we should avoid having references to specific things from general code, so instead this `is or or or` check I'd rather see a `!ProofObligation.UserControlled` (or a better name)